devito/propagator.py

```diff
@@ -29,29 +32,22 @@ class Propagator(object):
     :param block_size: Block size used for cache clocking
     """
     def __init__(self, name, nt, shape, spc_border=0, time_order=0,
-                 forward=True, compiler=None, profile=False,
-                 cache_blocking=False, block_size=5):
-        # Derive JIT compilation infrastructure
-        self.compiler = compiler or get_compiler_from_env()
-        self.mic_flag = isinstance(self.compiler, IntelMICCompiler)
-
-        self.t = symbols("t")
-        self.cache_blocking = cache_blocking
-        if(isinstance(block_size, Iterable)):
-            if(len(block_size) == len(shape)):
-                self.block_sizes = block_size
-            else:
-                raise ArgumentError("Block size should either be a single number or an array of the same size as the spatial domain")
-        else:
-            # A single block size has been passed. Broadcast it to a list of the size of shape
-            self.block_sizes = [int(block_size)]*len(shape)
-        # We assume the dimensions are passed to us in the following order
-        # var_order
-        if len(shape) == 2:
-            self.space_dims = symbols("x z")
-        else:
-            self.space_dims = symbols("x y z")
-        self.space_dims = self.space_dims[0:len(shape)]
+                 time_dim=None, space_dims=None, forward=True, compiler=None,
+                 profile=False, cache_blocking=False, block_size=5):
+        # Setting space order
+        self.spc_order = spc_border * 2
```

Comment:
Since `spc_order` seems to be a popular choice for a variable name, how about removing `spc_border` completely?